Trivago/Jade/Infrastructure/Repository/DoctrineResourceRepository.php

```diff
@@ -129,10 +130,26 @@ public function fetchResourcesWithConstraint(array $relationships, Constraint $c
         $idsConstraint = new Constraint();
         $idsConstraint->setPerPage(PHP_INT_MAX);
         $idBasedQueryBuilder = $this->createQuery($prefix, $relationships, $idsConstraint, $sortCollection);
-        $idBasedQueryBuilder->where($prefix.'.id IN (:ids)')->setParameters(['ids' => $ids]);
+        $idBasedQueryBuilder->where($prefix . '.id IN (:ids)')->setParameters(['ids' => $ids]);
 
         $result = $idBasedQueryBuilder->getQuery()->execute();
 
+        // Verifying if ARRAY_CONTAINS or ARRAY_NOT_CONTAINS was used
+        // and filter the results
+        $arrContainsFilters = $this->getArrayContainsFilters($constraint->getFilterCollection());
```

Comment:
I'm really sorry but let's move this code to another PR
It has a major issue and like this I can't merge your first feature
The main issue with this approach is the following
Row 1: (apple2)
Row 2: (apple)
I search for apple with limit 1
What I get in your current approach is an empty list since the first row is fetched by doctrine and later on removed by your array_filter

Reply:
OK. I will put each feature in a separate branch and on different PR.